rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java

```diff
@@ -129,6 +131,33 @@ public ResponseEntity<?> createVendor(
         return ResponseEntity.created(location).body(halResource);
     }
 
+    @Operation(
+            summary = "Update a vendor.",
+            description = "Update a vendor.",
+            tags = {"Vendor"}
+    )
+    @PreAuthorize("hasAuthority('WRITE')")
+    @RequestMapping(value = VENDORS_URL + "/{id}", method = RequestMethod.PATCH)
+    public ResponseEntity<?> updateVendor(
+            @Parameter(description = "The id of the vendor")
+            @PathVariable("id") String id,
+            @Parameter(description = "The vendor to be updated.")
+            @RequestBody Vendor vendor
+    ) {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        if (vendor.getFullname() == null && vendor.getShortname() == null && vendor.getUrl() == null) {
```

Comment:
Shouldn't these conditions be `||` instead of `&&` to prevent reading and updating `null` values for some fields?

Reply:
If I use || condition then it will return bad request whenever user wants to update specific fields.

Reply:
Make sense